docs/source/starter/style_guide.rst

```diff
@@ -147,26 +153,33 @@ In practice, this code looks like:
 
 Forward vs training_step
 ========================
-We recommend using forward for inference/predictions and keeping training_step independent
+
+We recommend using forward for inference/predictions and keeping ``training_step`` independent.
 
 .. code-block:: python
 
     def forward(self, x):
         embeddings = self.encoder(x)
+        return embeddings
 
 
-    def training_step(self):
-        x, y = ...
+    def training_step(self, batch, batch_idx):
+        x, y = batch
         z = self.encoder(x)
         pred = self.decoder(z)
         ...
 
-However, when using DataParallel, you will need to call forward manually
+However, when using DataParallel, you will need to call forward manually:
 
 .. code-block:: python
 
-    def training_step(self):
-        x, y = ...
+    def forward(self, x):
+        embeddings = self.encoder(x)
+        return embeddings
+
+
+    def training_step(self, batch, batch_idx):
+        x, y = batch
         z = self(x)  # < ---------- instead of self.encoder(x)
         pred = self.decoder(z)
```

Comment:
a question in general (even though you did not change this): If we need to call `self(x)` instead of `self.encoder(x)`, can we still call `self.decoder(z)` or would this also have to go to `forward`? I don't see a difference there. If there is one, it probably has to be explained here...

Reply:
yes.. this isn't required anymore. I think.
since batch is already splitter at this point, so calling self(), or self.encoder(x) won't be a problem of any sort.

for DP model is warpped around `model=DataParallel(LightningParallelModule(LightningModule)))`

`LightningParallelModule` takes care of which method to call, while `DataParallel` splits the batch and calls LightningParallelModule's forward which then calls `training_step` while training with the splitted batch.

cc: @akihironitta

Reply:
this is what I could find from my investigation. But if anyone can confirm this, I'll update this.

Reply:
I think so as well :)

Reply:
thanks @justusschock for bringing this up :)